packages/scripts/src/remove-training-cases.js

```diff
@@ -0,0 +1,354 @@
+	const { id: caseId, reference } = caseDetails;
+
+	const folders = await tx.folder.findMany({
+		where: { caseId }
+	});
+
+	const documents = await tx.document.findMany({
+		where: { caseId }
+	});
+
+	const removeFoldersByParentId = async (folderId, folderPath = reference) => {
+		const childFolders = folders.filter((folder) => folder.parentFolderId === folderId);
+
+		for (const folder of childFolders) {
+			await removeFoldersByParentId(folder.id, `${folderPath}/${folder.displayNameEn}`);
+		}
+
+		if (folderId) {
+			const folderDocuments = documents.filter((document) => document.folderId === folderId);
+
+			for (const document of folderDocuments) {
+				await removeDocument(tx, document, folderPath);
+			}
+
+			console.log(`Removing folder: ${folderPath}`);
+
+			await tx.folder.delete({ where: { id: folderId } });
+		}
+	};
+
+	await removeFoldersByParentId(null);
+};
+
+/**
+ *
+ * @param {PrismaClient} tx
+ * @param caseDetails
+ * @returns {Promise<void>}
+ */
+const removeOtherAssociatedRecords = async (tx, caseDetails) => {
+	const { id: caseId, reference } = caseDetails || {};
+	console.log(`Removing case status: ${reference}`);
+	await tx.caseStatus.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing published state: ${reference}`);
+	await tx.casePublishedState.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing grid reference: ${reference}`);
+	await tx.gridReference.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing project team: ${reference}`);
+	await tx.projectTeam.deleteMany({
+		where: { caseId }
+	});
+	console.log(`Removing subscription: ${reference}`);
+	await tx.subscription.deleteMany({
+		where: { caseId }
+	});
+};
+
+/**
+ * @param {string} reference
+ * @returns {Promise<void>}
+ */
+const removeCase = async (reference) => {
+	try {
+		const startTime = Date.now();
+		await databaseConnector.$transaction(
+			async (tx) => {
+				const { id: caseId } = (await getCaseByRef(reference)) || {};
+				if (!caseId) {
+					throw new Error('No such case ' + reference);
+				}
+				const caseDetails = await getCaseById(caseId, {
+					applicationDetails: true,
+					applicant: true
+				});
+
+				await removeApplicationDetails(tx, caseDetails);
+				await removeExaminationTimetables(tx, caseDetails);
+				await removeRepresentations(tx, caseDetails);
+				await removeProjectUpdates(tx, caseDetails);
+				await removeS51Advices(tx, caseDetails);
+				await removeOtherAssociatedRecords(tx, caseDetails);
+				await removeFoldersAndDocuments(tx, caseDetails);
+				await removeApplicant(tx, caseDetails);
+				await removeS51Advices(tx, caseDetails);
+
+				await tx.case.delete({ where: { id: caseId } });
+
+				console.log(`${reference} Removed in ${(Date.now() - startTime) / 1000} seconds`);
+
+				if (isSimulatedTest) {
+					throw new Error('Simulated deletion test');
+				}
+			},
+			{
+				maxWait: 20000, // default: 2000
+				timeout: 100000 // default: 5000
```

Comment:
Is it worth making these configurable with environment variables?

Reply:
Done